src/api/source/mvt.js

```diff
@@ -0,0 +1,452 @@
+        const d2t = perpendicular(d2);
+
+        const s = dot(sub(o2, o1), d2t) / dot(d1, d2t);
+        const t = dot(sub(o1, o2), d1t) / dot(d2, d1t);
+
+        if (s >= 0 && s <= 1 && t >= 0 && t <= 1) {
+            // Intersects!
+            return [o1[0] + s * d1[0], o1[1] + s * d1[1]];
+        }
+        // Doesn't intersects
+    }
+
+    _decodeLines(geom, featureGeometries, mvt_extent) {
+        let geometry = [];
+        geom.map(l => {
+            let line = [];
+            l.map(point => {
+                line.push(2 * point.x / mvt_extent - 1, 2 * (1 - point.y / mvt_extent) - 1);
+            });
+            geometry.push(line);
+        });
+        featureGeometries.push(geometry);
+    }
+
+    _decodeMVTLayer(mvtLayer, mvt_extent, MNS, jsonFields, numFields, stringFields, fieldMap) {
+        const properties = [];
+        this.metadata.columns.map(c => {
+            var e = null;
+            if (c.type == 'number') {
+                e = new Float32Array(mvtLayer.length + 1024);
+            } else if (c.type == 'json') {
+                e = {};
+            } else if (c.type == 'category') {
+                e = {};
+            }
+            properties[fieldMap[c.name]] = e;
+        });
+        var points = null;
+        const geomType = VectorTileFeature.types[mvtLayer.feature(0).type].toLowerCase();
+        let featureGeometries = [];
+        for (var i = 0; i < mvtLayer.length; i++) {
+            const f = mvtLayer.feature(i);
+            const geom = f.loadGeometry();
+            if (geomType == 'point') {
+                if (points == null) {
+                    points = new Float32Array(mvtLayer.length * 2);
+                }
+                points[2 * i + 0] = 2 * (geom[0][0].x) / mvt_extent - 1.;
+                points[2 * i + 1] = 2 * (1. - (geom[0][0].y) / mvt_extent) - 1.;
+            } else if (geomType == 'polygon') {
+                this._decodePolygons(geom, featureGeometries, mvt_extent);
+            } else if (geomType == 'line') {
+                this._decodeLines(geom, featureGeometries, mvt_extent);
+            } else {
+                throw new Error(`Unimplemented geometry type: '${geomType}'`);
+            }
+
+            // TODO check number of properties received and send error if not all the defined
+            // properties are being received
+            this.metadata.columns.map(c => {
+                var e = null;
+                if (c.type === 'number') {
+                    e = Number(f.properties[c.name]);
+                } else if (c.type == 'json') {
+                    e = JSON.parse(f.properties[c.name]);
+                } else if (c.type == 'category') {
+                    e = f.properties[c.name];
+                }
+                properties[fieldMap[c.name]][i] = e;
+            });
+        }
+
+        return { properties, points, featureGeometries, geomType };
+    }
+
+    _computePointGeometry(data) {
```

Comment:
I think all these `compute*` functions are dead code, right? @

Reply:
deleted, you're right